File: ios/Cookpit/Cookpit/ExploreViewModel.swift
//
//  ExploreViewModel.swift
//  Cookpit
//
//  Created by Kittinun Vantasin on 6/1/16.
//  Copyright © 2016 Cookpit. All rights reserved.
//

import Foundation

enum ExploreViewModelCommand {

  case SetItems(items: [CPExploreDetailViewData])
  
}

struct ExploreViewModel {

  let items: [CPExploreDetailViewData]
  
  func executeCommand(command: ExploreViewModelCommand) -> ExploreViewModel {
    switch command {
    case let .SetItems(items):
        return ExploreViewModel(items: items)
    }
  }
  
}

File: android/Cookpit/app/src/main/kotlin/com/github/kittinunf/cookpit/explore/ExploreViewModel.kt
package com.github.kittinunf.cookpit.explore

import com.github.kittinunf.cookpit.ExploreDetailViewData

sealed class ExploreViewModelCommand {

    class SetItems(val items: List<ExploreDetailViewData> = listOf()) : ExploreViewModelCommand()

}

data class ExploreViewModel(val items: List<ExploreDetailViewData> = listOf()) {

    fun executeCommand(command: ExploreViewModelCommand): ExploreViewModel {
        when(command) {
            is ExploreViewModelCommand.SetItems -> {
                return ExploreViewModel(command.items)
            }
        }
    }

}

